cli/main.go

```diff
@@ -128,6 +157,76 @@ func main() {
 		EnableBashCompletion:   true,
 		UseShortOptionHandling: true,
 		Commands: []*cli.Command{
+			{
+				Name:  "stale-balance",
+				Args:  true,
+				Usage: "If needed, calls `verifyStaleBalances` to correct the balance on a pod you don't own. This will attempt to (1) conclude any existing checkpoint on the pod, and then (2) invoke verifyStaleBalances() to start another, more up-to-date, checkpoint.",
+				Flags: []cli.Flag{
+					POD_ADDRESS_FLAG,
+					EXEC_NODE_FLAG,
+					BEACON_NODE_FLAG,
+					Require(SENDER_PK_FLAG),
+					SLASHED_VALIDATOR_INDEX_FLAG,
+					NO_PROMPT_FLAG,
+					BatchBySize(&batchSize, DEFAULT_BATCH_CHECKPOINT),
+				},
+				Action: func(cctx *cli.Context) error {
+					ctx := context.Background()
+
+					eth, beacon, chainId, err := core.GetClients(ctx, node, beacon)
+					core.PanicOnError("failed to get clients", err)
+
+					ownerAccount, err := core.PrepareAccount(&sender, chainId)
+					core.PanicOnError("failed to parse sender PK", err)
+
+					eigenpod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), eth)
+					core.PanicOnError("failed to reach eigenpod", err)
+
+					currentCheckpointTimestamp, err := eigenpod.CurrentCheckpointTimestamp(nil)
+					core.PanicOnError("failed to fetch any existing checkpoint info", err)
+
+					if currentCheckpointTimestamp > 0 {
+						// TODO: complete current checkpoint
+						fmt.Printf("This eigenpod has an outstanding checkpoint (since %d). You must complete it before continuing.", currentCheckpointTimestamp)
+
+						proofs, err := core.GenerateCheckpointProof(ctx, eigenpodAddress, eth, chainId, beacon)
+						core.PanicOnError("failed to generate checkpoint proofs", err)
+
+						txns, err := core.SubmitCheckpointProof(ctx, sender, eigenpodAddress, chainId, proofs, eth, batchSize, noPrompt)
+						core.PanicOnError("failed to submit checkpoint proofs", err)
+
+						for i, txn := range txns {
+							fmt.Printf("sending txn[%d/%d]: %s (waiting)...", i, len(txns), txn.Hash())
+							bind.WaitMined(ctx, eth, txn)
+						}
+					}
```

Comment:
This could be quite expensive - should we add a flag that signals that you want to complete an existing checkpoint if it's there?

Reply:
like i'm picturing a worst-case scenario where we run the health checker on Renzo's pods and out of 8k validators one got slashed

and then we auto-complete a checkpoint for them

which, at 15 gwei, is ballpark ~6/7 ETH. Then, we start a checkpoint for them and complete a second one, for another 6/7 ETH :sweat:

Reply:
`submitCheckpointProof` has warning messages all over it (see how we pass `noPrompt` through). We should indicate ahead of time that _starting_ the checkpoint could lead to finishing an expensive one though..

but that's sort of the same problem we face on checkpoints rn in the CLI -- we don't do any cost/gas estimation yet